batch/batch/front_end/front_end.py

```diff
@@ -2139,12 +2187,21 @@ async def ui_get_job(request, userdata, batch_id):
 
     # Not all logs will be proper utf-8 but we attempt to show them as
     # str or else Jinja will present them surrounded by b''
-    job_log_strings_or_bytes = {}
-    for container, log in job_log_bytes.items():
-        try:
-            job_log_strings_or_bytes[container] = log.decode('utf-8') if log is not None else None
-        except UnicodeDecodeError:
-            job_log_strings_or_bytes[container] = log
+    job_log_strings_or_bytes: Dict[str, Union[str, bytes, None]] = {}
+    possibly_truncated_logs = set()
+    for container, log_stream in job_log_streams.items():
+        if log_stream is None:
+            job_log_strings_or_bytes[container] = None
+        else:
+            fifty_kib = 50 * 1024
+            async with log_stream:
+                log = await log_stream.read(fifty_kib)
+            if len(log) == fifty_kib:
+                possibly_truncated_logs.add(container)
```

Comment:
```suggestion
            fifty_kib = 50 * 1024
            async with log_stream:
                log = await log_stream.read(fifty_kib + 1)
            if len(log) == fifty_kib + 1:
                possibly_truncated_logs.add(container)
            log = log[:fifty_kib]
```

Let's give the user confidence.